pkg/controller/controller.go

```diff
@@ -27,20 +27,32 @@ import (
 	"github.com/abcxyz/jvs/internal/project"
 	"github.com/abcxyz/jvs/pkg/justification"
 	"github.com/abcxyz/pkg/renderer"
+
 	"golang.org/x/exp/slices"
 	"google.golang.org/protobuf/types/known/durationpb"
 )
 
+// processor is the mockable interface for the convenience of testing.
+type processor interface {
+	CreateToken(context.Context, string, *jvspb.CreateJustificationRequest) ([]byte, error)
+	Validators() map[string]jvspb.Validator
+}
```

Comment:
I think you can still use the existing processor? You just create a processor with validators.

Reply:
Processor is from another package and requires keyManagementClient and justificationConfig to construct a new instance, so it is easier to mock it out to avoid all those complexity 

https://github.com/abcxyz/jvs/blob/c74e145411363aed1d245d5530ed92f246df1abe/pkg/justification/processor.go#L54

Reply:
Likely you can simply do

```go
p := NewProcessor(nil, &config.JustificationConfig{}).WithValidators(...)
```

Reply:
Actually followed Seth's suggestion from https://github.com/abcxyz/jvs/pull/300#discussion_r1260028123 to precompute category and pairs in constructor, turned out mockProcessor is not needed any longer. Already cleaned up the PR, PTAL